tsfile/src/main/java/org/apache/iotdb/tsfile/read/common/RowRecord.java

```diff
@@ -36,14 +41,31 @@ public RowRecord(long timestamp) {
   public RowRecord(long timestamp, List<Field> fields) {
     this.timestamp = timestamp;
     this.fields = fields;
+    for (Field field : fields) {
+      if (field == null || field.getDataType() == null) {
+        hasNullField = true;
+      } else {
+        allNull = false;
+      }
+    }
```

Comment:
If we doesn't use the `without null` clause, we also have to check each field again, which may have an impact on performance.